test/booking_manager_test.rb

```diff
@@ -0,0 +1,345 @@
+
+      expect(@booking_manager.blocks.length).must_equal 1
+    end
+
+    it "raises an exception if the total rooms requested exceeds the number of available rooms" do
+      @booking_manager.book_single_res("06-05-2020", "06-10-2020")
+
+      expect{@booking_manager.create_block(
+        check_in: "06-09-2020",
+        check_out: "06-12-2020",
+        total_rooms: 5,
+        group_name: "SAA",
+        discount: 0.10
+      )}.must_raise ArgumentError
+    end
+
+    it "raises an exception if the total rooms requested exceeds 5" do
+      expect{@booking_manager.create_block(
+        check_in: "06-09-2020",
+        check_out: "06-12-2020",
+        total_rooms: 6,
+        group_name: "SAA",
+        discount: 0.10
+      )}.must_raise ArgumentError
+    end
+
+    it "cannot create a block with a room that's part of another block" do
+      @booking_manager.create_block(
+        check_in: "07-01-2020",
+        check_out: "07-03-2020",
+        total_rooms: 3,
+        group_name: "SAA",
+        discount: 0.10
+      )
+      @booking_manager.create_block(
+        check_in: "07-01-2020",
+        check_out: "07-03-2020",
+        total_rooms: 2,
+        group_name: "SCA",
+        discount: 0.10
+      )
+
+      block1_rooms = @booking_manager.blocks[0].rooms.map! {|room|
+        room.number}
+      block2_rooms = @booking_manager.blocks[1].rooms.map! {|room|
+        room.number}
+
+      expect(block1_rooms).wont_include block2_rooms[0]
+      expect(block1_rooms).wont_include block2_rooms[1]
+    end
+  end
+
+  describe "find block" do
+    before do
+      @booking_manager.create_block(
+        check_in: "10-10-2020", 
+        check_out: "10-15-2020", 
+        total_rooms: 3, 
+        group_name: "Ramoz", 
+        discount: 0.10
+      )
+    end
+
+    it "can find a block given a group name" do
+      found_block = @booking_manager.find_block("Ramoz")
+
+      expect(found_block).must_equal @booking_manager.blocks[0]
+    end
+
+    it "returns nil if no blocks found" do
+      found_block = @booking_manager.find_block("SAA")
+
+      expect(found_block).must_be_nil
+    end
+  end
+
+  describe "book block res" do
+    before do 
+      @booking_manager.create_block(
+        check_in: "10-10-2020", 
+        check_out: "10-15-2020", 
+        total_rooms: 3, 
+        group_name: "Ramoz", 
+        discount: 0.10
+      )
+
+      @block_res = @booking_manager.blocks[0]
+    end
+
+    it "shortens the block's number of unreserved rooms by 1" do
+      @booking_manager.book_block_res("Ramoz")
+      unreserved_room_nums = @block_res.unreserved_rooms.map {|room| room.number}
+      
+      expect(@booking_manager.blocks.length).must_equal 1
+      expect(@block_res.unreserved_rooms.length).must_equal 2
+      expect(@block_res.rooms.length).must_equal 3
+      expect(unreserved_room_nums).wont_include 1
+      expect(unreserved_room_nums).must_include 2 && 3
```

Comment:
Hm, I'm not sure that this actually makes Ruby happy. I may be wrong, but afaik, you'll need to separate the `must_include 2` and `must_include 3` assertions